### PR TITLE
Feature/slack integration

### PR DESCRIPTION
**Pull request by tomasinchoo**
Hi everyone,

Here is slack integration and purge all functionality.
@vvuksan You can test https://github.com/Inchoo/fastly-drupal/tree/feature/slack-integration just in case.

### CODE REVIEW

**Comment by tomasinchoo** [comment]
@vvuksan

**Comment by vvuksan** [comment]
Did you mean to close this ?

File: src/Api.php [hunk 1]
```diff
@@ -223,6 +234,21 @@ public function purgeUrl($url = '') {
   }
 
   /**
+   * Performs an actual purge request for the given path.
+   */
+  protected function purgeQuery($path) {
+    drupal_http_request(url($path, ['absolute' => TRUE]), [
```
Comment (greylabel) [inline src/Api.php]: `drupal_http_request ` is not part of the Drupal 8 API : https://api.drupal.org/api/drupal/includes%21common.inc/function/drupal_http_request/7.x

File: src/Api.php [hunk 2]
```diff
@@ -67,6 +76,8 @@ public function __construct(ConfigFactoryInterface $config_factory, $host, Clien
     $this->httpClient = $http_client;
     $this->logger = $logger;
     $this->state = $state;
+    $this->_webhook = $webhook;
+    $this->_base_url = \Drupal::request()->getHost();
```
Comment (greylabel) [inline src/Api.php]: This should be injected rather than using `\Drupal::` since we are in a service.

File: src/Api.php [hunk 3]
```diff
@@ -16,6 +17,8 @@
  */
 class Api {
 
+  protected $_base_url;
```
Comment (greylabel) [inline src/Api.php]: I don't think starting the variable name with `_` follows coding standards. https://www.drupal.org/docs/develop/standards/coding-standards#naming

**Comment by tomasinchoo** [comment]
Hi everyone, I must have been closed request accedentaly, but please don't merge it before I go through comments.

File: src/Api.php (hunk 1 above)
Comment (tomasinchoo) [reply src/Api.php]: That's something that was already there. I only added $this->_webhook->sendWebHook('Purged ' . $path, "purge_keys"); 
purgeQuery is not actually used anywhere as I see, so I'll just remove it.

File: src/Api.php (hunk 3 above)
Comment (tomasinchoo) [reply src/Api.php]: I'll change that. D8 standards are joke though :)

**Comment by tomasinchoo** [comment]
Hi everyone,
I created changes that @greylabel suggested.

**Pull request merged**